contracts/axelar-gateway/tests/auth.rs

```diff
@@ -149,7 +149,10 @@ fn rotate_signers_fail_empty_signers() {
     let proof = generate_proof(&env, data_hash, signers);
 
     assert_contract_err!(
-        client.try_rotate_signers(&empty_signers, &proof, &true),
+        invoke_auth!(
```

Comment:
what's the point of using this macro? This test isn't checking auth, but a specific contract failure, so auth check isn't really relevant here, and can be checked separately

Reply:
it's more useful for successful executions where we want to ensure that the call would NOT fail due to an auth issue as well. So your usage of `assert_invoke_auth_ok` later is useful

Reply:
Use `client.mock_all_auths().try_rotate_signers(&empty_signers, &proof, &true)` instead to explicitly mock it. This is more precise than the generic `env.mock_all_auths()` so it doesn't ignore legitimate auth checks